test/resume/TcpResumeServer.cpp

```diff
@@ -0,0 +1,185 @@
+ public:
+  explicit ServerSubscription(Subscriber<Payload>& response)
+      : response_(response) {}
+
+  ~ServerSubscription(){};
+
+  // Subscription methods
+  void request(size_t n) override {
+    response_.onNext(Payload("from server"));
+    response_.onNext(Payload("from server2"));
+    response_.onComplete();
+    //    response_.onError(std::runtime_error("XXX"));
+  }
+
+  void cancel() override {}
+
+ private:
+  Subscriber<Payload>& response_;
+};
+
+class ServerRequestHandler : public DefaultRequestHandler {
+ public:
+  /// Handles a new inbound Subscription requested by the other end.
+  void handleRequestSubscription(Payload request, Subscriber<Payload>& response)
+      override {
+    LOG(INFO) << "ServerRequestHandler.handleRequestSubscription " << request;
+
+    response.onSubscribe(createManagedInstance<ServerSubscription>(response));
+  }
+
+  /// Handles a new inbound Stream requested by the other end.
+  void handleRequestStream(Payload request, Subscriber<Payload>& response)
+      override {
+    LOG(INFO) << "ServerRequestHandler.handleRequestStream " << request;
+
+    response.onSubscribe(createManagedInstance<ServerSubscription>(response));
+  }
+
+  void handleFireAndForgetRequest(Payload request) override {
+    LOG(INFO) << "ServerRequestHandler.handleFireAndForgetRequest " << request
+              << "\n";
+  }
+
+  void handleMetadataPush(std::unique_ptr<folly::IOBuf> request) override {
+    LOG(INFO) << "ServerRequestHandler.handleMetadataPush "
+              << request->moveToFbString() << "\n";
+  }
+
+  void handleSetupPayload(ConnectionSetupPayload request) override {
+    LOG(INFO) << "ServerRequestHandler.handleSetupPayload " << request << "\n";
+  }
+};
+
+class Callback : public AsyncServerSocket::AcceptCallback {
+ public:
+  Callback(EventBase& eventBase, Stats& stats)
+      : eventBase_(eventBase), stats_(stats){};
+
+  virtual ~Callback() = default;
+
+  virtual void connectionAccepted(
+      int fd,
+      const SocketAddress& clientAddr) noexcept override {
+    std::cout << "connectionAccepted" << clientAddr.describe() << "\n";
+
+    auto socket =
+        folly::AsyncSocket::UniquePtr(new AsyncSocket(&eventBase_, fd));
+
+    std::unique_ptr<DuplexConnection> connection =
+        folly::make_unique<TcpDuplexConnection>(std::move(socket), stats_);
+    std::unique_ptr<DuplexConnection> framedConnection =
+        folly::make_unique<FramedDuplexConnection>(std::move(connection));
+    std::unique_ptr<RequestHandler> requestHandler =
+        folly::make_unique<ServerRequestHandler>();
+
+    std::unique_ptr<ReactiveSocket> rs = ReactiveSocket::fromServerConnection(
+        std::move(framedConnection),
+        std::move(requestHandler),
+        stats_,
+        [&](ReactiveSocket& socket, const ResumeIdentificationToken& token)
+        {
+            std::cout << "resume callback" << std::endl;
+            socket.resumeFromSocket(*reactiveSockets_[0]);
```

Comment:
Due to the way the API works and the internal objects, we need a new ReactiveSocket on the server side (from fromServerConnection) to resume on. i.e. a new connection comes in and we don't know until a RESUME or SETUP is seen if it will be a resume or not. So, the old ReactiveSocket is an old session. And the new ReactiveSocket is a resumed session. Which means we need to move the streams and resume state to the new ReactiveSocket.

Also, the user only has ReactiveSocket to associate with a Resume Token.